src/liballoc/collections/btree/map.rs

```diff
@@ -1804,6 +1877,41 @@ impl<'a, K, V> RangeMut<'a, K, V> {
             }
         }
     }
+
+    // Transform given leaf edge handle into handles of next kv and next leaf edge
+    fn next_handles(
```

Comment:
I put this here as an associated function because it matches the bulk of next_unchecked (except the decomposition into k,v and the fact that it's not unchecked)